mne/viz/evoked.py

```diff
@@ -1500,7 +1501,12 @@ def plot_compare_evokeds(evokeds, picks=list(), gfp=False, colors=None,
     # are pulled from the 'colors' and 'linestyles' dicts via '/'-tag matching
     # unless they are overwritten by entries from a user-provided 'styles'.
 
-    # first, check if input is valid
+    # first, copy to avoid overwriting
+    for style_dict in [styles, colors, linestyles]:
+        if isinstance(style_dict, dict):
+            style_dict = deepcopy(style_dict)
```

Comment:
> I don't think a new test is necessary?

This is always a bit dangerous. Hopefully you at least verified that it works locally? 

...because I suspect that these three lines do nothing to protect `styles`, `colors`, and `linestyles`, instead replacing a local variable `style_dict` with a `deepcopy`:
```
>>> from copy import deepcopy
>>> a = dict()
>>> b = dict()
>>> c = dict()
>>>
>>>
>>> def foo(styles, colors, linestyles):
>>>     for style_dict in [styles, colors, linestyles]:
>>>         if isinstance(style_dict, dict):
>>>             style_dict = deepcopy(style_dict)
>>>     styles['foo'] = 'bar'
>>>     colors['a'] = 'b'
>>>     linestyles['c'] = 'd'
>>>
>>> print(a)
{}
>>> foo(a, b, c)
>>> print(a)
{'foo': 'bar'}
```
You could [go out of your way to do it programmatically] (http://stackoverflow.com/questions/5036700/how-can-you-dynamically-create-variables-via-a-while-loop) but the result will probably not be that readable, so maybe use the three lines of the form `styles = deepcopy(styles)` in the right conditional branches.